pkg/engine/imageVerify.go

```diff
@@ -451,7 +463,6 @@ func (iv *imageVerifier) verifyAttestorSet(
 	imageVerify kyvernov1.ImageVerification,
 	imageInfo apiutils.ImageInfo,
 	path string,
-	predicateType string,
```

Comment:
How predicateType is used to verify attestations.attestors if we remove it here?

Reply:
In the prior implementation, even though the `predicateType` was passed to `verifyAttestorSet` it was not used, except when a recursive call to `verifyAttestorSet` was made for nested attestors.

The predicateType is used in the `verifyAttestations` method:
- it gets passed to the buildOptionsAndPath method here https://github.com/kyverno/kyverno/pull/5713/files/5a236506126cb947968c6c394306888ac150d843#diff-d752e78274f7472fae120f3f51e21b4aa62c25162a295be400a771e190fe9bfdR421 which passes it to Cosign here https://github.com/JimBugwadia/kyverno/blob/5a236506126cb947968c6c394306888ac150d843/pkg/engine/imageVerify.go#L575.
- it gets passed to verifyAttestion here https://github.com/JimBugwadia/kyverno/blob/5a236506126cb947968c6c394306888ac150d843/pkg/engine/imageVerify.go#L575 which checks the attestations for the type https://github.com/JimBugwadia/kyverno/blob/5a236506126cb947968c6c394306888ac150d843/pkg/engine/imageVerify.go#L641.

Reply:
>even though the predicateType was passed to verifyAttestorSet it was not used,

It was used to build and verify attestors if not empty:

https://github.com/kyverno/kyverno/blob/ef63302dc4794c6f8a9ed8c695c9eb613b0abb7d/pkg/engine/imageVerify.go#L439